### DISCUSSION TITLE
Can ch_PP-OCRv4_rec_server_infer's support for english be put into the documentation?

### DISCUSSION THREAD

**Discussion by ryx2** [PaddlePaddle/PaddleOCR#12317] [Q&A]
I notice if I am calling

```
from paddleocr import PaddleOCR
ocr = Paddle.OCR(
det_model_dir=ch_PP-OCRv4_det_server_infer,
rec_model_dir=ch_PP-OCRv4_rec_infer
lang='en')
...
result = ocr.ocr(my_image)
```
this works fine. However, If i set the rec model to the server version as well (`ch_PP-OCRv4_rec_server_infer`), then I get the following error:

```
  File "/opt/conda/lib/python3.10/site-packages/paddleocr/paddleocr.py", line 661, in ocr
    dt_boxes, rec_res, _ = self.__call__(img, cls)
  File "/opt/conda/lib/python3.10/site-packages/paddleocr/tools/infer/predict_system.py", line 105, in __call__
    rec_res, elapse = self.text_recognizer(img_crop_list)
  File "/opt/conda/lib/python3.10/site-packages/paddleocr/tools/infer/predict_rec.py", line 628, in __call__
    rec_result = self.postprocess_op(preds)
  File "/opt/conda/lib/python3.10/site-packages/paddleocr/ppocr/postprocess/rec_postprocess.py", line 121, in __call__
    text = self.decode(preds_idx, preds_prob, is_remove_duplicate=True)
  File "/opt/conda/lib/python3.10/site-packages/paddleocr/ppocr/postprocess/rec_postprocess.py", line 83, in decode
    char_list = [
  File "/opt/conda/lib/python3.10/site-packages/paddleocr/ppocr/postprocess/rec_postprocess.py", line 84, in <listcomp>
    self.character[text_id]
IndexError: list index out of range
```

Which I'm guessing is because it's trying to output Chinese, which has an 8000 character dict, whereas English only has 90 or so. Because it says english is supported by the server model (https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.7/doc/doc_ch/models_list.md), is it possible to get the ppocrv4 server model to output english successfully? 
<img width="1274" alt="Screen Shot 2024-03-11 at 10 12 15 PM" src="https://github.com/PaddlePaddle/PaddleOCR/assets/21298347/f0b204ea-c7d3-4368-a939-4c9f99b111fb">

**Comment by TingquanGao** [comment]
please try to set lang to 'ch', or use the en_PP-OCRv4_rec model.

**Comment by ryx2** [comment]
So what you're saying is that `ch_PP-OCRv4_rec_server_infer` doesn't actually support English?

**Comment by tink2123** [comment]
`ch_PP-OCRv4_rec_server_infer` supports Chinese + English. 
lang=en supports pure English, "en" and "ch" correspond to different dictionaries respectively. 

https://github.com/PaddlePaddle/PaddleOCR/blob/69832ab5326c6db614af6fb74b530aeae1c9b80e/paddleocr.py#L93-L96

https://github.com/PaddlePaddle/PaddleOCR/blob/69832ab5326c6db614af6fb74b530aeae1c9b80e/paddleocr.py#L88-L91

Model and dictionary need to be consistent, so when you use `ch_PP-OCRv4_rec_server_infer` , do not modify the lang parameters.

**Comment by ryx2** [comment]
Ah I see now, Chinese + English != {Chinese, English}, got it
